core/rid.h

```diff
@@ -92,63 +101,41 @@ class RID_OwnerBase {
 		p_rid._data = p_data;
 		refcount.ref();
 		p_data->_id = refcount.get();
-#ifndef DEBUG_ENABLED
 		p_data->_owner = this;
-#endif
 	}
 
-#ifndef DEBUG_ENABLED
+	_FORCE_INLINE_ void _free(RID &p_rid) {
+		p_rid._data = nullptr;
+	}
 
```

Comment:
This name seems confusing, you aren't freeing anything, you are setting a pointer to NULL.

Reply:
`RID_OwnerBase::_free()` is only called by `RID_Owner::free()`, which is the current API in both 3.x and 4.x. What name would you suggest?